pkg/generate/app/sourcelookup.go

```diff
@@ -269,7 +269,8 @@ func (r *SourceRepository) RemoteURL() (*url.URL, bool, error) {
 		if len(ref) > 0 {
 			remote = fmt.Sprintf("%s#%s", remote, ref)
 		}
-		if r.remoteURL, err = url.Parse(remote); err != nil {
+
+		if r.remoteURL, err = git.ParseRepository(remote); err != nil {
```

Comment:
there you go ... i had thought either @csrwng or I had made this change in new-app, but nope ... good catch